src/Stack/Options/GlobalParser.hs

```diff
@@ -90,6 +94,12 @@ globalOptsFromMonoid defaultTerminal GlobalOptsMonoid{..} = do
     , globalStylesUpdate = globalMonoidStyles
     , globalTermWidth = getFirst globalMonoidTermWidth
     , globalStackYaml = stackYaml
+    , globalLockFileBehavior =
+        let defLFB =
+              case (getFirst globalMonoidResolver, getFirst globalMonoidCompiler) of
```

Comment:
Does compiler override change anything in this case?
If we talk about getting different packages in a source map then extra targets on command line could more easily give you a different source map though neigther `LFBReadOnly` nor `LFBReadWrite` look meaningful in this case

Reply:
Maybe I'm being overly cautious about the compiler. I think you're right: I can't think of anything that would cause lock files to behave differently there.

I was under the (perhaps mistaken) impression that extra targets on the command line would not affect the source map here. If they do, you're right, that's the bigger concern. Is there an easy way at this stage of processing to know whether there are extra targets?

Reply:
It appears that it was me who got confused by different source maps that we have :)
In `SMWanted` we don't have any targets yet so it looks like resolver is the only option that could cause problems here.